service/matching/taskListManager.go

```diff
@@ -562,26 +562,35 @@ func (c *taskListManagerImpl) CancelPoller(pollerID string) {
 
 // Returns a batch of tasks from persistence starting form current read level.
 // Also return a number that can be used to update readLevel
-func (c *taskListManagerImpl) getTaskBatch() ([]*persistence.TaskInfo, int64, error) {
+func (c *taskListManagerImpl) getTaskBatch() ([]*persistence.TaskInfo, int64, bool, error) {
 	var tasks []*persistence.TaskInfo
 	readLevel := c.taskAckManager.getReadLevel()
 	maxReadLevel := c.taskWriter.GetMaxReadLevel()
+	times := 0
+	isReadBatchDone := true
 	for readLevel < maxReadLevel {
 		upper := readLevel + c.config.RangeSize
 		if upper > maxReadLevel {
 			upper = maxReadLevel
 		}
 		tasks, err := c.getTaskBatchWithRange(readLevel, upper)
 		if err != nil {
-			return nil, readLevel, err
+			return nil, readLevel, isReadBatchDone, err
 		}
 		// return as long as it grabs any tasks
 		if len(tasks) > 0 {
-			return tasks, upper, nil
+			return tasks, upper, isReadBatchDone, nil
 		}
 		readLevel = upper
+
+		times++
+		if times >= 100 {
```

Comment:
Let's add code to return back every 10 iteration.

Reply:
Changed